src/System.Transactions/src/System/Transactions/VolatileEnlistmentMultiplexing.cs

```diff
@@ -0,0 +1,524 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Transactions.Diagnostics;
+
+namespace System.Transactions
+{
+    // The volatile Demultiplexer is a fanout point for promoted volatile enlistments.
+    // When a transaction is promoted a single volatile enlistment is created in the new
+    // transaction for all volitile enlistments on the transaction.  When the VolatileDemux
+    // receives a preprepare it will fan that notification out to all of the enlistments
+    // on the transaction.  When it has gathered all of the responses it will send a 
+    // single vote back to the DistributedTransactionManager.
+    internal abstract class VolatileDemultiplexer : IEnlistmentNotificationInternal
+    {
+        // Reference the transactions so that we have access to it's enlistments
+        protected InternalTransaction _transaction;
+        
+        // Store the IVolatileEnlistment interface to call back to the Distributed TM
+        internal IPromotedEnlistment _oletxEnlistment;
```

Comment:
>oletx [](start = 38, length = 5)

We should remove all usage of "oletx" in identifiers like this. In this particular case, it should be _promotedEnlistment. In other cases, "distributed" could be used as a replacement. But we might as well get rid of references to oletx now and make the identifiers more general.